src/frontend.md

```diff
@@ -5,31 +5,56 @@
 ```
 src/
   components/ (can be dumb and smart)
-    bankAccount/
+    [All components should be placed in some entity folder or shared folder]
+    bankAccount/ (entity, module, .etc)
       BankAccountTable/
         [no nested directories]
+        [avoid index.ts]
         BankAccountTable.ts
         BankAccountTable.test.ts
         BankAccountTableBadge.ts (child component)
         BankAccountTable.styled.ts
+        BankAccountTable.hook.ts (ui hook used only in the current component)
```

Comment:
hooks can be ui-less

```suggestion
        BankAccountTable.hooks.ts (ui/logic hooks used only in the current component)
```

Reply:
ui-less hooks should be placed in queries or services

Reply:
if i have several large hooks, can i have several files with names like `useBankAccountTable.ts`, `useBankAccountTableWhatever.ts`?

Reply:
@viczhuravlev if it's supposed to be used in only one component, why we should put it to another folder?

Reply:
@Rendalf 
if there is one or two hooks, then I agree that it is unnecessary.
If more, it will help keep the code clean.

Reply:
@viczhuravlev what do you mean by `keep the code clean`? could you give us some guide about clean code?

Reply:
> if i have several large hooks, can i have several files with names like `useBankAccountTable.ts`, `useBankAccountTableWhatever.ts`?

Of course, I wrote `BankAccountTable.hook.ts` just to emphasize is a file with hook inside.
Maybe to avoid misunderstanding better to rename to `useBankAccountTable.ts`?
And I think we can put several local hooks to component folder.

Reply:
> @Rendalf if there is one or two hooks, then I agree that it is unnecessary. If more, it will help keep the code clean.

I think, generally, components are not so complicated to need a lot of hooks and folder `hooks` for them. If the component has a lot of local hooks then maybe better to split it to several components.